### ISSUE TITLE
paddle稀疏反卷积算子支持需求

### ISSUE THREAD

**Issue by YanxianChen** [PaddlePaddle/Paddle#73308]
### 需求描述 Feature Description

任务目标（请描述你正在做的项目是什么，如模型、论文、项目是什么？）; <br /> 需求场景（请描述你的项目中为什么需要用此功能）;  <br /> 功能描述（请简单描述或设计这个功能）
目前我们在paddle上复现CVPR2024 点云检测相关任务（HEDNet，SAFDNet，github链接：https://github.com/zhanggang001/HEDNet/tree/main/pcdet），论文中有一个关键的Encoder-Decoder结构需要用到稀疏反卷积操作，目前paddle还不支持，辛苦增加这部分算子。

**Comment by yuanlehome** [comment]
您好，已经反馈给相关同事，有消息再同步～

**Comment by jeff41404** [comment]
> [https://github.com/zhanggang001/HEDNet/tree/main/pcdet），论文中有一个关键的Encoder-Decoder结构需要用到稀疏反卷积操作，目前paddle还不支持，辛苦增加这部分算子。](https://github.com/zhanggang001/HEDNet/tree/main/pcdet%EF%BC%89%EF%BC%8C%E8%AE%BA%E6%96%87%E4%B8%AD%E6%9C%89%E4%B8%80%E4%B8%AA%E5%85%B3%E9%94%AE%E7%9A%84Encoder-Decoder%E7%BB%93%E6%9E%84%E9%9C%80%E8%A6%81%E7%94%A8%E5%88%B0%E7%A8%80%E7%96%8F%E5%8F%8D%E5%8D%B7%E7%A7%AF%E6%93%8D%E4%BD%9C%EF%BC%8C%E7%9B%AE%E5%89%8Dpaddle%E8%BF%98%E4%B8%8D%E6%94%AF%E6%8C%81%EF%BC%8C%E8%BE%9B%E8%8B%A6%E5%A2%9E%E5%8A%A0%E8%BF%99%E9%83%A8%E5%88%86%E7%AE%97%E5%AD%90%E3%80%82)

你好，确认下这里希望提供的“稀疏反卷积”的具体接口形式，比如当前飞桨稀疏卷积的接口形式为：
[稀疏2D卷积官网文档](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/sparse/nn/Conv2D_cn.html)
`class paddle.sparse.nn.Conv2D(in_channels, out_channels, kernel_size, stride=1, padding=0, dilation=1, groups=1, padding_mode='zeros', weight_attr=None, bias_attr=None, data_format='NHWC')`
[稀疏3D卷积官网文档](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/sparse/nn/Conv3D_cn.html)
`class paddle.sparse.nn.Conv3D(in_channels, out_channels, kernel_size, stride=1, padding=0, dilation=1, groups=1, padding_mode='zeros', weight_attr=None, bias_attr=None, data_format='NDHWC')`

同时，飞桨当前提供了一些[稀疏计算接口paddle.sparse](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/sparse/Overview_cn.html)，我们也欢迎继续新增接口，如果可以的话希望能贡献代码到飞桨，[具体新增接口的方法可参考此链接](https://www.paddlepaddle.org.cn/documentation/docs/zh/dev_guides/api_contributing_guides/api_contributing_guides_cn.html)

**Comment by YanxianChen** [comment]
这块spconv代码库里有pytorch版的第三方实现，https://github.com/traveller59/spconv/blob/master/spconv/pytorch/conv.py#L1077，可以作为参考～。对应到paddlepaddle的算子接口，可以是如下的格式，可以按实际开发情况做调整
2D反卷积：
class paddle.sparse.nn.InverseConv2d(in_channels, out_channels, kernel_size, weight_attr=None, bias_attr=None, data_format='NHWC')
3D反卷积：
class paddle.sparse.nn.InverseConv3D(in_channels, out_channels, kernel_size, weight_attr=None, bias_attr=None, data_format='NDHWC')
